torch/csrc/jit/function.h

```diff
@@ -87,20 +88,24 @@ struct TORCH_API Function {
 
   GraphExecutor& get_executor() {
     ensure_defined();
-    std::call_once(executor_init_, [&] {
-      check_single_output();
-      executor_ = GraphExecutor(graph());
-    });
+    if (executor_) {
+      return executor_;
+    }
+    std::lock_guard<std::mutex> lock(compile_mutex);
```

Comment:
this is a race condition if get_executor gets called from multiple threads - the read from executor_ should be guarded by the lock too